R/register_extendr.R

```diff
@@ -72,27 +81,59 @@ make_wrappers <- function(module_name, package_name, outfile,
   )
   x <- stringi::stri_split_lines1(x)
 
+  # Can't use usethis::write_over because it asks user for input if
+  # file already exists.
+  brio::write_lines(x, outfile)
   if (!isTRUE(quiet)) {
-    message("Writing wrappers to:\n", outfile)
+    rel_path <- pretty_rel_path(outfile, search_from = path)
+    cli::cli_alert_success("Writting wrappers to {.file {rel_path}}.")
   }
-  brio::write_lines(x, outfile)
+}
+
+# Checks if new wrappers should be generated
+needs_new_warppers <- function(path = ".", wrapper_path = fs::path("R", "extendr-wrappers.R")) {
+  wrapper_path <- rprojroot::find_package_root_file(wrapper_path, path = path)
+
+  if (!fs::file_exists(wrapper_path)) {
+    # No wrappers, they should be generated
+    return(TRUE)
+  }
+
+  # Retrieves path to e.g. 'src/my_package.dll'
+  library_path <- get_library_path(path)
+
+  if (!fs::file_exists(library_path)) {
+    # No library found
+    cli::cli_alert_danger("Library file {.file {pretty_rel_path(library_path)}} is missing, cannot generate wrappers!")
+    stop("Wrapper generation failed. Aborting.", call. = FALSE)
+  }
+
+  wrapper_info <- fs::file_info(wrapper_path)
+  library_info <- fs::file_info(library_path)
+
+  # If wrapeprs are older than the library file, new wrappers are needed.
+  library_info[["modification_time"]] > wrapper_info[["modification_time"]]
 }
 
 make_wrappers_externally <- function(module_name, package_name, outfile,
-                                     use_symbols = FALSE, quiet = FALSE) {
-  func <- function(package_root, make_wrappers, ...) {
-    pkgload::load_all(package_root, quiet = TRUE)
+                                     use_symbols = FALSE, quiet = FALSE,
+                                     path = ".", compile = NA) {
+  func <- function(package_root, make_wrappers, compile, ...) {
+    pkgload::load_all(package_root, compile = compile, quiet = TRUE)
     make_wrappers(...)
   }
 
   args <- list(
-    package_root = rprojroot::find_package_root_file(path = "."),
+    package_root = rprojroot::find_package_root_file(path = path),
     make_wrappers = make_wrappers,
+    compile = compile,
+    # arguments passed to make_wrappers()
     module_name = module_name,
     package_name = package_name,
     outfile = outfile,
     use_symbols = use_symbols,
-    quiet = quiet
+    quiet = quiet,
+    path = path
```

Comment:
Sorry, probably this should be `rprojroot::find_package_root_file(path = path)` as well.

Reply:
Yeah I found a few places, if it works locally I will push the updates.